### PR TITLE
Now search history is stored in state and can be traversed using arrow keys while focus is on search text

### PR DESCRIPTION
**Pull request by saurabh95**
This implementation doesn't have any UI as per this PR. We can have a discussion regarding its UI. There is also a trello card regarding the same with UI also https://trello.com/c/IzACddEv/344-s-find-recent-search-history .
Will have UI implementation(as per the discussion) in next PR.

### CODE REVIEW

File: src/search/FindBar.js [hunk 1]
```diff
@@ -310,9 +311,21 @@ define(function (require, exports, module) {
...
+                var maxCount = PreferencesManager.get("maxSearchHistory");
                 if (e.keyCode === KeyEvent.DOM_VK_RETURN) {
                     e.preventDefault();
                     e.stopPropagation();
+                    var searchQueryIndex = searchHistory.indexOf($('#find-what').val());
+                    if (searchQueryIndex !== -1) {
+                        searchHistory.splice(searchQueryIndex, 1);
+                    } else {
+                        if (searchHistory.length === maxCount) {
+                            searchHistory.splice(maxCount - 1, 1);
```
Comment (zaggino) [inline src/search/FindBar.js]: isn't `.pop()` the same thing?

File: src/search/FindBar.js [hunk 2]
```diff
@@ -333,6 +346,15 @@ define(function (require, exports, module) {
                         // if Shift is held down).
                         self.trigger("doFind", e.shiftKey);
                     }
+                    currentIndex = 0;
+                } else if (e.keyCode === KeyEvent.DOM_VK_DOWN) {
+                    currentIndex = (currentIndex - 1 + Math.min(maxCount, searchHistory.length)) % Math.min(maxCount, searchHistory.length);
```
Comment (zaggino) [inline src/search/FindBar.js]: line too long (140 chars)

File: src/search/FindBar.js [hunk 3]
```diff
@@ -267,6 +267,7 @@ define(function (require, exports, module) {
         FindBar._addFindBar(this);
 
         var $root = this._modalBar.getRoot();
+        var currentIndex = 0;
```
Comment (zaggino) [inline src/search/FindBar.js]: I'd rename this to `historyIndex` or something little more specific

File: src/search/FindBar.js [hunk 4]
```diff
@@ -333,6 +346,15 @@ define(function (require, exports, module) {
                         // if Shift is held down).
                         self.trigger("doFind", e.shiftKey);
                     }
+                    currentIndex = 0;
+                } else if (e.keyCode === KeyEvent.DOM_VK_DOWN) {
+                    currentIndex = (currentIndex - 1 + Math.min(maxCount, searchHistory.length)) % Math.min(maxCount, searchHistory.length);
+                    $("#find-what").val(searchHistory[currentIndex]);
+                    self.trigger("queryChange");
+                } else if (e.keyCode === KeyEvent.DOM_VK_UP) {
+                    currentIndex = (currentIndex + 1 + Math.min(maxCount, searchHistory.length)) % Math.min(maxCount, searchHistory.length);
```
Comment (zaggino) [inline src/search/FindBar.js]: can you wrap `if (e.keyCode === KeyEvent.DOM_VK_DOWN)` and `if (e.keyCode === KeyEvent.DOM_VK_UP) {` codes into a function taking an argument up or down? code seems to repeat itself

**Review by zaggino — APPROVED** [review]

**Review by swmitra — APPROVED** [review]
Implementation looks good to me. We might want to add some UI at a later point of time.

**Comment by swmitra** [comment]
@saurabh95 Great job! This is a very handy feature for developers. Keep these coming... 👍

**Pull request merged**